rules-emerging-threats/2023/Malware/Pikabot/net_connection_win_malware_pikabot_rundll32_activity.yml

```diff
@@ -0,0 +1,34 @@
+title: Suspicious Process Invocation by Rundll32 - Possibly Pikabot C2 activity
+id: cae6cee6-0244-44d2-84ed-e65f548eb7dc
+status: experimental
+description: |
+    Detects execution of rundll32 that leads to an external network connection.
+    The malware Pikabot have been seen to use this technique to initiate C2-communication through hard-coded Windows binaries.
+references:
+    - https://www.virustotal.com/gui/file/d72af640b71b8e3eca3eba660dd7c7f029ff8852bcacaa379e7b6c57cf4d9b44
+author: Andreas Braathen (mnemonic.io)
+date: 2023/10/27
+tags:
+    - attack.command_and_control
+    - attack.t1573
+    - detection.emerging_threats
+logsource:
+    product: windows
+    category: network_connection
+detection:
+    selection_parent:
+        ParentName: 'rundll32.exe'
```

Comment:
Network connection event do not have the Parent Image by default, so it can't be used here

Reply:
Depends on the backend, e.g. MDE does have parentname (not image) on network events, which makes this analytic valid. How would you propose to solve this? 

Curious of the Sigma strategy here, because it would be less optimal if one could not take advantage of such data fields.

https://learn.microsoft.com/en-us/microsoft-365/security/defender/advanced-hunting-devicenetworkevents-table?view=o365-worldwide

Reply:
After some internal discussions. We decided to include this for now. I would prefer to use a similar taxonomy to what we have (sysmon inspired). I'll make the changes.

As for the long explanation. Since these are very specific rules, meaning they use fields that are only available to some. It would defy the purpose to add them with the rest of the rule as it'll be confusing to users. Without providing full context on how to leverage the rule.

Its basically writing an MDE or Crowdstrike rule but just in Sigma.

In the future we will add a dedicated folder for these kinds of rules to make it clear for users that these are 3rd party/extended rules that are using somewhat unique fields